OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseTabsDashboardFragment.java

```diff
@@ -95,30 +97,36 @@ public static CourseTabsDashboardFragment newInstance(Bundle bundle) {
     public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
         courseData = (EnrolledCoursesResponse) requireArguments().getSerializable(Router.EXTRA_COURSE_DATA);
+        binding = FragmentCourseTabsDashboardBinding.inflate(inflater, container, false);
 
         if (courseData != null) {
             setHasOptionsMenu(courseData.getCourse().getCoursewareAccess().hasAccess());
             environment.getAnalyticsRegistry().trackScreenView(
                     Analytics.Screens.COURSE_DASHBOARD, courseData.getCourse().getId(), null);
-
             if (!courseData.getCourse().getCoursewareAccess().hasAccess()) {
                 final boolean auditAccessExpired = new Date().after(DateUtil.convertToDate(courseData.getAuditAccessExpires()));
-                FragmentDashboardErrorLayoutBinding errorLayoutBinding =
-                        FragmentDashboardErrorLayoutBinding.inflate(inflater, container, false);
-                errorLayoutBinding.errorMsg.setText(auditAccessExpired ? R.string.course_access_expired : R.string.course_not_started);
-                return errorLayoutBinding.getRoot();
+                if (auditAccessExpired) {
+                    setupToolbar();
+                    binding.toolbar.tabs.setVisibility(View.GONE);
+                    binding.accessError.getRoot().setVisibility(View.VISIBLE);
+                    binding.accessError.primaryButton.setOnClickListener(onFindCourseClick());
+                } else {
+                    FragmentDashboardErrorLayoutBinding errorLayoutBinding = FragmentDashboardErrorLayoutBinding.inflate(inflater, container, false);
+                    errorLayoutBinding.errorMsg.setText(R.string.course_not_started);
+                    return errorLayoutBinding.getRoot();
+                }
             } else {
-                binding = FragmentCourseTabsDashboardBinding.inflate(inflater, container, false);
                 setupToolbar();
                 setViewPager();
-                return binding.getRoot();
             }
+            return binding.getRoot();
         } else if (getArguments().getBoolean(ARG_COURSE_NOT_FOUND)) {
             // The case where we have invalid course data
-            FragmentDashboardErrorLayoutBinding errorLayoutBinding =
-                    FragmentDashboardErrorLayoutBinding.inflate(inflater, container, false);
-            errorLayoutBinding.errorMsg.setText(R.string.cannot_show_dashboard);
-            return errorLayoutBinding.getRoot();
+            binding.loadingError.getRoot().setVisibility(View.VISIBLE);
```

Comment:
better to define the state too. like `binding.stateLayout.state.setState(EdxErrorState.State.LOAD_ERROR)`